userscripts/install-synology-dsm7.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+# Notifiarr DSM 7+ Installer
+
+set -e
+
+# Check for root privileges
+if [ "$EUID" -ne 0 ]; then
+  echo "This script must be run as root or with sudo."
+  exit 1
+fi
+
+echo "Detecting system architecture..."
+ARCH=$(uname -m)
+case "$ARCH" in
+  x86_64 | amd64) 
+    ARCH_NAME="amd64"
+    ;;
+  aarch64 | arm64) 
+    ARCH_NAME="arm64"
+    ;;
+  armv7* | armv6* | armhf) 
+    ARCH_NAME="arm"
+    ;;
+  *) 
+    echo "Unsupported architecture: $ARCH"
+    exit 1
+    ;;
+esac
+
+echo "Fetching latest Notifiarr release metadata..."
+RELEASE_JSON=$(curl -sSL https://api.github.com/repos/Notifiarr/notifiarr/releases/latest)
+PACKAGE_URL=$(echo "$RELEASE_JSON" | grep -o "https://[^\"]*notifiarr\.${ARCH_NAME//./\\.}\.linux\.gz" | head -n 1)
+PACKAGE=$(basename "$PACKAGE_URL")
+
+if [ -z "$PACKAGE_URL" ]; then
+  echo "Failed to find a valid release package for $ARCH_NAME"
+  exit 1
+fi
+
+echo "Downloading Notifiarr package: $PACKAGE_URL"
+cd /tmp
+if ! curl -L -o "$PACKAGE" "$PACKAGE_URL"; then
+  echo "Failed to download Notifiarr package."
+  exit 1
+fi
+
+echo "Installing or updating Notifiarr binary..."
+if [ -f /usr/bin/notifiarr ]; then
+  echo "Stopping existing Notifiarr service..."
+  pkill -f "/usr/bin/notifiarr" || true
+  # Give processes time to terminate
+  sleep 2
+fi
+
+echo "Extracting Notifiarr package..."
+if ! gunzip -c "$PACKAGE" > /usr/bin/notifiarr; then
+  echo "Failed to extract Notifiarr package."
+  exit 1
+fi
+
+chmod +x /usr/bin/notifiarr
+rm -f "$PACKAGE"
+
+echo "Creating or updating config and log directories..."
+mkdir -p /etc/notifiarr /volume1/@appdata/notifiarr/lognotifiarr
+CONFIGFILE=/etc/notifiarr/notifiarr.conf
+if [ ! -f "${CONFIGFILE}" ]; then
+  echo "Generating config file ${CONFIGFILE}"
+  echo " " > "${CONFIGFILE}"
+  export DN_LOG_FILE="/volume1/@appdata/notifiarr/lognotifiarr/app.log"
+  export DN_HTTP_LOG="/volume1/@appdata/notifiarr/lognotifiarr/http.log"
+  /usr/bin/notifiarr --config "${CONFIGFILE}" --write "${CONFIGFILE}.new"
+  mv "${CONFIGFILE}.new" "${CONFIGFILE}"
+else
+  echo "Config file already exists. Skipping generation."
+fi
+
+echo "Creating 'notifiarr' user if missing..."
+if ! id notifiarr >/dev/null 2>&1; then
+  synouser --add notifiarr "" Notifiarr 0 "" 0
+fi
+
+echo "Setting permissions/ownership on: /usr/bin/notifiarr /volume1/@appdata/notifiarr/lognotifiarr"
+chmod 0750 /usr/bin/notifiarr /volume1/@appdata/notifiarr/lognotifiarr
+chown -R notifiarr:notifiarr /volume1/@appdata/notifiarr/lognotifiarr /etc/notifiarr
+
+echo "Adding sudoers entry for smartctl..."
+sed -i '/notifiarr/d' /etc/sudoers
```

Comment:
This will remove anything else with this word. I usually don't remove, and just use grep to check if it exists already before adding it with echo.